dhcpv4/dhcpv4.go

```diff
@@ -737,6 +737,13 @@ func (d *DHCPv4) MaxMessageSize() (uint16, error) {
 	return GetUint16(OptionMaximumDHCPMessageSize, d.Options)
 }
 
+// AutoConfigure returns the AutoConfigure option if present.
+//
+// The AutoConfigure option is described by RFC 2563, Section 2.
+func (d *DHCPv4) AutoConfigure() (byte, error) {
```

Comment:
I would define a type for this.

```
type AutoConfiguration byte

const (
  DoNotAutoConfigure AutoConfiguration = 0
  AutoConfigure AutoConfiguration = 1
)
```

AutoConfigure should return that.

It should have a stringer that prints "do not configure" or "configure" so that when Message.String() is used it doesn't just print "0" or "1"

Reply:
I wouldn't mind us using generics to make this nicer at some point, but if you're just defining this one, let's do this.